conn/gpio/gpiotest/gpiotest.go

```diff
@@ -128,5 +129,60 @@ func (p *PinPWM) PWM(duty gpio.Duty, period time.Duration) error {
 	return nil
 }
 
+// since returns time in µs since the test start.
+func since(start time.Time) string {
+	µs := (time.Since(start) + time.Microsecond/2) / time.Microsecond
+	ms := µs / 1000
+	µs %= 1000
+	return fmt.Sprintf("%3d.%03dms", ms, µs)
+}
+
+// LogPin logs when its state changes.
+type LogPin struct {
+	gpio.PinIO
+	start time.Time
```

Comment:
I'm not sure this is a good point here, as in general users just want to log and if they need timestamp, they shall use log.SetFlags(Lmicroseconds).
https://golang.org/pkg/log/#SetFlags